src/main/webapp/app/exercises/programming/manage/update/update-components/programming-exercise-difficulty.component.ts

```diff
@@ -17,4 +17,14 @@ export class ProgrammingExerciseDifficultyComponent {
     protected readonly ProjectType = ProjectType;
 
     faQuestionCircle = faQuestionCircle;
+
+    updateReadOnly() {
```

Comment:
```suggestion
    updateReadOnly() {
        this.programmingExercise.allowOnlineEditorReadOnlyMode = !this.isReadOnlyDisabled();
    }
```
The name `updateReadOnly` doesn't make sense to me if the flag can only be set to `false`. I'd expect the behavior of my suggestion.
If that's not correct consider renaming the method.

Reply:
Thanks for your feedback. The behavior that you suggested is not entirely what I wanted. The idea behind this function is to remove the check from the read-only option when conditions aren't satisfied anymore. The read-only checkbox can only be checked if both offline IDE and online editor are enabled. I renamed the function to make it more descriptive.